scarlet/source.py

```diff
@@ -78,112 +111,123 @@ def __init__(self, model_frame, sky_coord, observations):
+            observations = (observations,)
 
-        # initialize from observation
-        morph, thresh_starlet, bbox = init_starlet_source(
-            sky_coord,
+        # initialize morphology from model_frame psf
+        morph, bbox = self.init_morph(sky_coord, model_frame)
+        center = model_frame.get_pixel(sky_coord)
+        morphology = ExtendedSourceMorphology(
             model_frame,
-            observations,
-            spectrum=spectrum,
-            coadd=coadd,
-            coadd_rms=coadd_rms,
-            symmetric=False,
+            center,
+            morph,
+            bbox=bbox,
             monotonic="angle",
-            min_grad=min_grad,
-            full=full,
-            starlet_thresh=starlet_thresh,
+            symmetric=False,
+            min_grad=0,
+            shifting=shifting,
         )
 
-        spectrum = TabulatedSpectrum(model_frame, spectrum)
-        morphology = StarletMorphology(model_frame,
-                                       morph,
-                                       threshold=thresh_starlet,
-                                       bbox=bbox,
-                                       )
+        # get spectrum from peak pixel, correct for PSF
+        spectra = init.get_pixel_spectrum(sky_coord, observations, correct_psf=True)
+        spectrum = np.concatenate(spectra, axis=0)
+        spectrum /= morph.sum()
+        noise_rms = np.concatenate(
+            [np.array(np.mean(obs.noise_rms, axis=(1, 2))) for obs in observations]
+        ).reshape(-1)
+        spectrum = TabulatedSpectrum(model_frame, spectrum, min_step=noise_rms)
 
+        # set up model with its parameters
         super().__init__(model_frame, spectrum, morphology)
 
+        # retain center as attribute
+        self.center = morphology.center
+
+    def init_morph(self, sky_coord, frame):
+        """Initialize a source just like `init_extended_morphology`,
+        but with the morphology of a point source.
+        """
+
+        # position in frame coordinates
+        center = frame.get_pixel(sky_coord)
+        center_index = np.round(center).astype("int")
+
+        # morphology initialized as a point source
+        morph_ = frame.psf.get_model().mean(axis=0)
+        origin = (
+            center_index[0] - (morph_.shape[0] // 2),
+            center_index[1] - (morph_.shape[1] // 2),
+        )
+        bbox_ = Box(morph_.shape, origin=origin)
+
+        # adjust box size to conform with extended sources
+        size = max(morph_.shape)
+        boxsize = init.get_minimal_boxsize(size)
+        morph = np.zeros((boxsize, boxsize))
+        origin = (
+            center_index[0] - (morph.shape[0] // 2),
+            center_index[1] - (morph.shape[1] // 2),
+        )
+        bbox = Box(morph.shape, origin=origin)
+
+        slices = overlapped_slices(bbox, bbox_)
+        morph[slices[0]] = morph_[slices[1]]
+
+        # apply max normalization
+        morph /= morph.max()
```

Comment:
That's exactly the opposite from what I've done here. I don't think spreading out init functions over multiple files is a solution, in particular because I have yet to see cases where those bits actually get reused for another source type than the one it was defined for.